src/ImageSharp/Processing/Processors/Transforms/RotateProcessor.cs

```diff
@@ -17,81 +16,129 @@ namespace SixLabors.ImageSharp.Processing.Processors
+        /// Gets or sets the angle of processMatrix in degrees.
         /// </summary>
-        public bool Expand { get; set; } = true;
+        public float Angle { get; set; }
+
+        /// <inheritdoc/>
+        protected override Matrix3x2 CreateProcessingMatrix()
+        {
+            if (this.transformMatrix == default(Matrix3x2))
+            {
+                this.transformMatrix = Matrix3x2Extensions.CreateRotationDegrees(-this.Angle, PointF.Empty);
+            }
+
+            return this.transformMatrix;
+        }
+
+        /// <inheritdoc/>
+        protected override void CreateNewCanvas(Rectangle sourceRectangle)
+        {
+            if (MathF.Abs(this.Angle) < Constants.Epsilon ||
+                MathF.Abs(this.Angle - 180) < Constants.Epsilon)
+            {
+                this.ResizeRectangle = sourceRectangle;
+            }
+
+            if (MathF.Abs(this.Angle - 90) < Constants.Epsilon ||
+                MathF.Abs(this.Angle - 270) < Constants.Epsilon)
+            {
+                // We always expand enumerated rectangle values
+                this.ResizeRectangle = new Rectangle(0, 0, sourceRectangle.Height, sourceRectangle.Width);
+            }
+
+            base.CreateNewCanvas(sourceRectangle);
+        }
 
         /// <inheritdoc/>
-        protected override void OnApply(ImageFrame<TPixel> source, Rectangle sourceRectangle, Configuration configuration)
+        protected override void OnApply(ImageFrame<TPixel> source, ImageFrame<TPixel> destination, Rectangle sourceRectangle, Configuration configuration)
         {
-            if (this.OptimizedApply(source, configuration))
+            if (this.OptimizedApply(source, destination, configuration))
             {
                 return;
             }
 
-            int height = this.CanvasRectangle.Height;
-            int width = this.CanvasRectangle.Width;
-            Matrix3x2 matrix = this.GetCenteredMatrix(source, this.processMatrix);
+            int height = this.ResizeRectangle.Height;
+            int width = this.ResizeRectangle.Width;
+            Matrix3x2 matrix = this.GetCenteredMatrix(source);
             Rectangle sourceBounds = source.Bounds();
 
-            using (var targetPixels = new PixelAccessor<TPixel>(width, height))
+            if (this.Sampler is NearestNeighborResampler)
             {
                 Parallel.For(
                     0,
                     height,
                     configuration.ParallelOptions,
                     y =>
-                    {
-                        Span<TPixel> targetRow = targetPixels.GetRowSpan(y);
+                        {
+                            Span<TPixel> destRow = destination.GetPixelRowSpan(y);
+
+                            for (int x = 0; x < width; x++)
+                            {
+                                var transformedPoint = Point.Rotate(new Point(x, y), matrix);
```

Comment:
It's actually a `SixLabors.Core` issue, but I noticed that `Point.Rotate()` is the same as `Point.Skew()`, the code should be moved into a single `Point.Transfrom()` method IMO, and we should add new overloads, accepting matrix by `ref` to reduce the copy operations.

I wonder if it's possible to do a better resampling strategy. Currently our weights are centered at a rounded position. I should really read more image processing literature.

Reply:
PR on Core submitted https://github.com/SixLabors/Core/pull/11